src/CommandingOfficers/DeployableCommander.java

```diff
@@ -102,13 +102,20 @@ public void modifyActionList(UnitContext uc)
   @Override
   public GameEventQueue receiveUnitDieEvent(Unit victim, XYCoord grave, Integer healthBeforeDeath)
   {
+    GameEventQueue events = super.receiveUnitDieEvent(victim, grave, healthBeforeDeath);
     // COUs die when they are killed
     if( COUs.contains(victim) )
     {
       COUsLost.add(victim);
-      onCOULost(victim);
+      GameEventQueue couLossEvents = onCOULost(victim);
+      if( null != couLossEvents )
+      {
+        if( null == events )
+          events = new GameEventQueue();
+        events.addAll(couLossEvents);
+      }
```

Comment:
Any reason `onCOULost` returns a `GameEventQueue` instead of just accepting one to push events into? This could avoid the null-checking here. I guess `ReceiveUnitDieEvent` already doesn't conform to that paradigm too.

Reply:
I guess taking a queue to use is maybe a more-obvious way of returning a list of side-effects.
That feels like bait for a larger refactor, though, yeah.

I've favored return values over out params a lot since it makes the function "more pure".
I guess that rule doesn't make as much sense when the function mutates state on its instance, so I'm down with making `GameEventListener` hooks returning `void`.
We can also throw the game or map in as a param for them too, to fix the thing in the thread above.

Reply:
> That feels like bait for a larger refactor, though, yeah

Maybe, maybe not. I already follow the "pass in the queue to populate" strategy in other (most?) places where we call functions that produce GameEvents.

> I've favored return values over out params a lot since it makes the function "more pure"

Not sure I buy that; I'd tend to think the method of receiving a function's output is a question of syntax rather than purity, and creating additional containers just to concatenate them is just inefficient (not that this is a driving concern for us).

Reply:
> > That feels like bait for a larger refactor, though, yeah
> 
> Maybe, maybe not. I already follow the "pass in the queue to populate" strategy in other (most?) places where we call functions that produce GameEvents.

Every listener hook returns `null` right now. Changing that's a fairly large refactor, I feel like.
> > I've favored return values over out params a lot since it makes the function "more pure"
> 
> Not sure I buy that; I'd tend to think the method of receiving a function's output is a question of syntax rather than purity, and creating additional containers just to concatenate them is just inefficient (not that this is a driving concern for us).

If I call `enqueueMoveEvent()` on the same event queue twice, I get different results than if it simply provided a queue as a return value. Calling the function changes the state of the process, and you would have to know the internals of `enqueueMoveEvent()` or redo any previous work to the queue to undo that change.

A "return-pattern `createMoveEvent()`" would allow me to produce the first's output (by adding each return value to my own queue immediately), but it isn't built into calling the function at all anymore.
Likewise, I could produce the output of the return-patterned one with the existing function by explicitly creating a new queue for each call to operate on.

I suppose it's _mostly_ semantics either way, but I do slightly prefer not modifying input variables directly since that means I can reset a given function multiple times during debugging without producing aberrant behavior.